src/worker/worker/collectors/base_web_collector.py

```diff
@@ -39,7 +39,8 @@ def parse_source(self, source):
         if additional_headers := source["parameters"].get("ADDITIONAL_HEADERS", None):
             self.update_headers(additional_headers)
         if user_agent := source["parameters"].get("USER_AGENT", None):
-            self.update_headers({"User-Agent": user_agent})
+            user_agent_dict = {"User-Agent": user_agent}
+            self.update_headers(json.dumps(user_agent_dict))
```

Comment:
Could be simplified with:

```python
self.headers.update({"User-Agent": user_agent})
```

Reply:
let's do that, this is unnecessarily complicated